src/charm.py

```diff
@@ -154,6 +173,44 @@ def main(self, event):
 
         self.model.unit.status = ActiveStatus()
 
+    def _on_sidepanel_relation_changed(self, event: RelationChangedEvent):
+        try:
+            self._check_leader()
+        except CheckFailed as check_failed:
+            self.model.unit.status = check_failed.status
+            return
+        if event.app.name in SIDEBAR_EXTRA_OPTIONS:
+            self.log.info(f"ADDING {event.app.name} to side panel")
+            side_bar_tabs_dict = json.loads(self._stored.side_bar_tabs)
+            if (
+                SIDEBAR_EXTRA_OPTIONS[event.app.name]["menuLink"]
+                not in side_bar_tabs_dict["menuLinks"]
+            ):
+                side_bar_tabs_dict["menuLinks"].append(
+                    SIDEBAR_EXTRA_OPTIONS[event.app.name]["menuLink"]
+                )
+                self.log.info(f"NEW SIDE BAR {side_bar_tabs_dict}")
+                self._stored.side_bar_tabs = json.dumps(side_bar_tabs_dict)
+                self.main(event)
+
+    def _on_sidepanel_relation_broken(self, event: RelationBrokenEvent):
+        try:
+            self._check_leader()
+        except CheckFailed as check_failed:
+            self.model.unit.status = check_failed.status
+            return
+        if event.app.name in SIDEBAR_EXTRA_OPTIONS:
+            self.log.info(f"REMOVING {event.app.name} to side panel")
+            side_bar_tabs_dict = json.loads(self._stored.side_bar_tabs)
+            side_bar_tabs_dict["menuLinks"] = [
+                link
+                for link in side_bar_tabs_dict["menuLinks"]
+                if link != SIDEBAR_EXTRA_OPTIONS[event.app.name]["menuLink"]
+            ]
+            self.log.info(f"NEW SIDE BAR {side_bar_tabs_dict}")
+            self._stored.side_bar_tabs = json.dumps(side_bar_tabs_dict)
```

Comment:
This is event-based (if relation leaves, remove relation from sidebar), which then means we need to pass the state of the sidebar between `_on_sidepanel_relation_broken` and `main`.  What about in main just doing something like
```python
def main():
  sidebar = compute_sidebar_from_relations()
  if sidebar != existing_configmap:
    update_configmap(sidebar)
  # then trigger pod restart
```

This removes the need for stored state and simplifies the event handling logic